### PR TITLE
[SF-608] Top Level Links

### PR DESCRIPTION
**Pull request by george-e-shaw-iv**
* Added the ability to set top level links when marshaling
JSON:API compliant responses.

### CODE REVIEW

File: request_test.go [hunk 1]
```diff
@@ -317,10 +317,7 @@ func TestUnmarshal_nonNumericID(t *testing.T) {
 }
 
 func TestUnmarshalSetsAttrs(t *testing.T) {
-	out, err := unmarshalSamplePayload()
-	if err != nil {
-		t.Fatal(err)
-	}
+	out := unmarshalSamplePayload(t)
```
Comment (jennylam-outreach) [inline request_test.go:320]: I'm curious to know why in this test code, we want to start ignoring the unmarshaling error.

Comment (jennylam-outreach) [reply request_test.go:320]: I see. the `func unmarshalSamplePayload` declaration has changed ([L643](https://github.com/getoutreach/jsonapi/pull/2/files#diff-d7ac8930f8073cd3703fe4775ba3b5f3a312c3ecd8a2ab8f6c0da5bb6764da13R634)).

Then I'm curious, in what situation do we want to return error, in what situation do we want to throw a failure within a function? For this matter, is there a guideline/pattern to follow based on commonly agreed best practices in Go?

**Review by jennylam-outreach — APPROVED** [review]
LGTM. Got one minor question for my education. 🙏

File: request_test.go (hunk 1 above)
Comment (george-e-shaw-iv) [reply request_test.go:320]: I replied to you in slack as to why it's not being checked anymore! It's a paradigm specific to go testing

**Pull request merged**